includes/rest/class-server.php

```diff
@@ -145,6 +146,55 @@ public static function authorize_activitypub_requests( $response, $handler, $req
 		return $response;
 	}
 
+	/**
+	 * Callback function to authorize each api requests
+	 *
+	 * @see WP_REST_Request
+	 *
+	 * @see https://www.w3.org/wiki/SocialCG/ActivityPub/Primer/Authentication_Authorization#Authorized_fetch
+	 * @see https://swicg.github.io/activitypub-http-signature/#authorized-fetch
+	 *
+	 * @param \WP_REST_Request $request  Request used to generate the response.
+	 *
+	 * @return bool Whether the request is authorized.
+	 */
+	public static function authorize_activitypub_permissions( $request ) {
```

Comment:
Do you think it would make sense to have a more specific name for that method, to indicate what it does exactly (check for `HEAD` and verify signature), and to separate for any future permission callback you may have for other endpoints, that will require different permissions?

Reply:
Totally, I just wanted to have something out, to start a discussion about if this even makes sense or if I miss something.

We had a discussion about that long time ago and we decided to use what we currently use and I can't remember the background of the decision 🫣